### ISSUE TITLE
docs: upgrade docs rollback procedure contains errors

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#11907]
1. We're currently discussing removing the "Restore system tables" section from e.g.
https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#restore-system-tables

Note, that it is currently not listed in the summary of https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#rollback-procedure
so if it is left around (though I doubt it will), it should be added to the summary.

2. In "Restore the configuration file", the following shell command is incorrect:
```
sudo cp -a /etc/scylla/scylla.yaml.backup-src | /etc/scylla/scylla.yaml
```

It should probably be:
```
sudo cp -a /etc/scylla/scylla.yaml.backup-src /etc/scylla/scylla.yaml
```

3. In "Gracefully shutdown ScyllaDB" there seems to be a leak of rst include directive into the pre-formatted text:
```
  nodetool drain
.. include:: /rst_include/scylla-commands-stop-index.rst
```

I'm not sure which version of the upgrade docs needs to fixed exactly since there are (too) many of those.
Do we have a "master" version we maintain that is the basis of the upgrade guides for each release (Redhat and Ubuntu/debian)?

**Comment by annastuchlik** [comment]
@bhalevy 
1. When can we expect the decision on whether this section should be removed? Which versions will it affect? Please let me know when it's decided what to do.
2. @roydahan Could you confirm that the command provided by @bhalevy is correct? Then, I can update it in the docs.
3. I'll fix it wherever necessary.

**Comment by roydahan** [comment]
The commands should be as the commands we test.
I'll assign someone to share them here.

**Comment by bhalevy** [comment]
> @bhalevy
> 
> 1. When can we expect the decision on whether this section should be removed? Which versions will it affect? Please let me know when it's decided what to do.
> 2. @roydahan Could you confirm that the command provided by @bhalevy is correct? Then, I can update it in the docs.
> 3. I'll fix it wherever necessary.

I think that we can safely remove the  "Restore system tables" section from 4.6 and on and correspondingly, from 2022.1 and on in enterprise.

@tgrabiec please confirm.

In any case, QA should test that first to make sure we didn't miss anything.

**Comment by tgrabiec** [comment]
@bhalevy I confirm.

**Comment by temichus** [comment]
@roydahan , @bhalevy 

I've checked that SCT doing in rolling-upgrade-centos7-test in rollback from 5.3.0~dev to 5.2.0

1) Restore system tables
this step is exists in test, but  performed after "[Reload systemd configuration](https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#reload-systemd-configuration)" and looks like:
```
bash /tmp/recover_system_tables.sh <snapshot_name>
```
where recover_system_tables.sh it is QA bash file that copied on node:
```
#!/bin/bash

for DIR in '/var/lib/scylla/data/system' '/var/lib/scylla/data/system_schema'; do
    for i in `sudo ls $DIR`;
    do
        sudo test -e $DIR/$i/snapshots/$1 && (sudo rm -f $DIR/$i/* || true) && sudo find $DIR/$i/snapshots/$1 -type f -exec sudo -u scylla /bin/cp {} $DIR/$i/ \;
    done
done

# Some new table may not exist in old version, `test -e` might failed (return -1),
# it might be return to this script.
true
```

2) Restore the configuration file
in SCT this step looks:
```
sudo cp /etc/scylla/scylla.yaml-backup /etc/scylla/scylla.yaml
```


3) Gracefully shutdown ScyllaDB
in SCT we also  do "snapshot" like in [Upgrade Steps](https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#drain-the-nodes-and-backup-the-data) and do " systemctl stop ":
```
nodetool drain
nodetool snapshot
sudo systemctl stop scylla-server.service
```

and some more diffs that I found:

1)Backup the configuration file in Upgrade Steps https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#backup-the-configuration-file
here SCT backups package also:
```
 node.remoter.run('sudo cp /etc/scylla/scylla.yaml /etc/scylla/scylla.yaml-backup')
            if node.is_rhel_like():
                node.remoter.run('sudo cp /etc/yum.repos.d/scylla.repo ~/scylla.repo-backup')
            else:
                node.remoter.run('sudo cp /etc/apt/sources.list.d/scylla.list ~/scylla.list-backup')
            backup_conf(node)
```

and restore in Rollback Procedure instead of [Remove the old repo file](https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#download-and-install-the-old-release) 
```
        if node.is_rhel_like():
            node.remoter.run('sudo cp ~/scylla.repo-backup /etc/yum.repos.d/scylla.repo')
            node.remoter.run('sudo chown root.root /etc/yum.repos.d/scylla.repo')
            node.remoter.run('sudo chmod 644 /etc/yum.repos.d/scylla.repo')
        else:
            node.remoter.run('sudo cp ~/scylla.list-backup /etc/apt/sources.list.d/scylla.list')
            node.remoter.run('sudo chown root.root /etc/apt/sources.list.d/scylla.list')
            node.remoter.run('sudo chmod 644 /etc/apt/sources.list.d/scylla.list')
```
 
2) Scylla [reinstallation](https://github.com/scylladb/scylladb/blob/master/docs/upgrade/_common/upgrade-guide-v4-rpm.rst#download-and-install-the-old-release) in Rollback Procedure also looks different:
Install:
```
sudo yum clean all
sudo rm -rf /var/cache/yum
sudo yum downgrade scylla-\*cqlsh -y
sudo yum remove scylla-\*cqlsh -y
sudo yum downgrade scylla\* -y
sudo yum install scylla -y
```
instead of
```
sudo yum clean all
sudo rm -rf /var/cache/yum
sudo yum remove scylla\\*tools-core
sudo yum downgrade scylla\\* -y
sudo yum install scylla
```

**Comment by temichus** [comment]
UPD
regarding 1. Restore system tables

it looks like we can  remove "Restore system tables" because rolling upgrade tests can be passed with the option
recover_system_tables: false 
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/artsiom_mishuta/job/5.2-rolling-upgrade-ubuntu22.04-test/

**Comment by bhalevy** [comment]
@annastuchlik can you please submit a fix to the documentation based on @temichus's input?

**Comment by bhalevy** [comment]
> @annastuchlik can you please submit a fix to the documentation based on @temichus's input?

ping @annastuchlik

**Comment by bhalevy** [comment]
ping~2 @annastuchlik

**Comment by bhalevy** [comment]
Cc @mykaul it seems like this issue falls between the chairs.
Who should own it?

**Comment by tarzanek** [comment]
@annastuchlik if Benny/Tomek say it's safe to be removed from 2022.1 (or 4.6) and on, then it still means OLDer versions need to keep it
and lots of people still run 4.6 or 2021 so they should get their howto exactly how you patched it for https://github.com/scylladb/scylla-enterprise/issues/3046

that said I also agree that you shall not delete anything until QA will provide a blessing that in newer Scyllas system tables are able to "rollback" automagically (are they really?)

**Comment by mykaul** [comment]
> @annastuchlik if Benny/Tomek say it's safe to be removed from 2022.1 (or 4.6) and on, then it still means OLDer versions need to keep it and lots of people still run 4.6 or 2021 so they should get their howto exactly how you patched it for [scylladb/scylla-enterprise#3046](https://github.com/scylladb/scylla-enterprise/issues/3046)

Neither 4.6  nor 2021 are supported. I wish we could remove all those outdated references.

> 
> that said I also agree that you shall not delete anything until QA will provide a blessing that in newer Scyllas system tables are able to "rollback" automagically (are they really?)

**Comment by temichus** [comment]
> @temichus Thanks for reviewing the PRs that fix the upgrade procedure in Open Source. Could you provide the same details as in [#11907 (comment)](https://github.com/scylladb/scylladb/issues/11907#issuecomment-1573549089) for Open Source-> Enterprise and Enterprise->Enterprise? What else needs to be fixed apart from removing the "Restore system tables"?

Sure, I put it in my queue
cc @kostja

**Comment by fruch** [comment]
FYI not doing this backup breaks upgrade from OSS to Enterprise

**Comment by fruch** [comment]
> @fruch Let me rephrase to make sure I'm following.
> 
> We should remove the "Restore system tables" step from the rollback procedure in the upgrade guides:
> 
> * From OSS to OSS (done)
> * From Enterprise to Enterprise (pending)
> 
> We should leave the "Restore system tables" step in the upgrade guides:
> 
> * From OSS to Enterprise.
> 
> Is that correct?

base on my finding on runs we had in 2023.1, yes (removing it on SCT, surfaced that, but it took us a while to notice)

**Comment by tarzanek** [comment]
while at this, upgrade docs should warn in big letters that no user schema changes or topology changes should be done during upgrades (since in such case when you'd alter during upgrade rollback to old sstables might even break the node)

**Comment by fruch** [comment]
> while at this, upgrade docs should warn in big letters that no user schema changes or topology changes should be done during upgrades (since in such case when you'd alter during upgrade rollback to old sstables might even break the node)

it's funny, cause @avikivity suggested we should remove that note.
because we are aiming to a situation upgrades can happen with user even know about it.

but we did seen a few cases we do inteintally change schemas during upgrade, and using raft consistent schema elevated some of the issues.

**Comment by temichus** [comment]
> bash /tmp/recover_system_tables.sh <snapshot_name>

@annastuchlik 
according to what I see in Jenkins jobs

**rollback** from 2023.1.3-20231219.b890271f125b to 2022.1.12-20231129.09932ad308 doing with recover_system_tables
from 2024.1.0~rc3-20240101.29b16c9e1ed0 to  2023.1.3-20231219.b890271f125b ( and 2022.2.16-20231215.b9e397d94d44) without  recover_system_tables
